main.c

```diff
@@ -93,6 +93,10 @@ int main(void) {
 }
 
 void setup() {
+	uint8_t retcode;         // variable to store retcode
+	uint8_t msg_str[40];     // variable to store a message string
+	uint8_t hmc_retcode; 	 // variable to store retcode hmc module
```

Comment:
You can use `retcode` and `msg_str` for HMC results too. So it's not needed to waste memory for unneeded variables.